File: 20.valid-parentheses.js
/*
 * @lc app=leetcode id=20 lang=javascript
 *
 * [20] Valid Parentheses
 */

// @lc code=start
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
  let change = true
  while (change) {
    let s2=s.replace(/(\(\)|\{\}|\[\])/,'')
    change = s2!==s
    s=s2
  }
  return !s.length
};
// @lc code=end

File: 20.valid-parentheses.py
#
# @lc app=leetcode id=20 lang=python3
#
# [20] Valid Parentheses
#

# @lc code=start
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        map = {
            ')': '(',
            '}': '{',
            ']': '[',
        }
        for c in s:
            if c not in map.keys():
                stack.append(c)
            else:
                if not len(stack) or stack.pop() != map[c]:
                    return False
        return not len(stack)
# @lc code=end

